gwcs/wcs/_wcs.py

```diff
@@ -430,7 +431,7 @@ def _call_backward(
         return result
 
     def outside_footprint(self, world_arrays):
```

Comment:
Instead of making direct modifications with NaNs it maybe better to create a mask on the world arrays so that no copies or modifications need to be made, and then evaluating the results using the mask.

Reply:
File an issue so we don't forget?